### PR TITLE
Add script that validate version fields in config.yaml

### PR DESCRIPTION
**Pull request by mik-laj**
We sometimes release new versions of Airflow with new options, but we forget to update the config.yml file. In order to detect such situations, I add a simple script that compares relalased config.yaml with local config.yaml , which allows semi-auttomatic validatte and updated the `version added` fields.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Comment by potiuk** [comment]
Should we make it as  pre-commit and fail if it does not validate? I am afraid if we leave it there in tools without actually running it, it will be quickly forgotten.

**Comment by potiuk** [comment]
Or at the very least - it should be part of dev/release procedure.

**Comment by mik-laj** [comment]
> Or at the very least - it should be part of dev/release procedure.

I added this to the release procedure. Good idea

File: dev/README_RELEASE_AIRFLOW.md [hunk 1]
```diff
@@ -745,3 +746,15 @@ EOF
 ## Update Announcements page
 
 Update "Announcements" page at the [Official Airflow website](https://airflow.apache.org/announcements/)
+
```
Comment (potiuk) [inline dev/README_RELEASE_AIRFLOW.md:749]: Should it be done "BEFORE" the release maybe?

File: scripts/tools/validate_version_added_fields_in_config.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
```
Comment (potiuk) [inline scripts/tools/validate_version_added_fields_in_config.py]: remove it from here :)

File: dev/README_RELEASE_AIRFLOW.md (hunk 1 above)
Comment (mik-laj) [reply dev/README_RELEASE_AIRFLOW.md:749]: We should keep the config.yml file in good shape both in the main branch and in the releasing branch. This script only allows the validation of files on the master branch. The `added_version` field on the releases branch is set to the correct values in branches while we cherry-pick commits.

Comment (potiuk) [reply dev/README_RELEASE_AIRFLOW.md:749]: I mean - right now the process is to update confiig.yaml - after the release is published (and documentation generated) - should not we simply run the script before we update version and prepare changelog (this is already happening in the release branch). This was my proposal - to move it at the very beginning of the file.

Comment (mik-laj) [reply dev/README_RELEASE_AIRFLOW.md:749]: I think we have two problems: 1. updating the file in the main branch to include information about the new release, and 2. making sure that the new options in the release branch contain the value in the `added_version` field.  I am taking care of the first problem for now.

Comment (potiuk) [reply dev/README_RELEASE_AIRFLOW.md:749]: Ah. I see. It's fetching from PYPI. fine for me then,

**Review by potiuk — APPROVED** [review]

File: scripts/tools/validate_version_added_fields_in_config.py (hunk 2 above)
Comment (potiuk) [reply scripts/tools/validate_version_added_fields_in_config.py]: Still this one should be removed :)

**Review by potiuk — APPROVED** [review]

File: scripts/tools/validate_version_added_fields_in_config.py (hunk 2 above)
Comment (mik-laj) [reply scripts/tools/validate_version_added_fields_in_config.py]: Done. I forgot to push commit.

**Pull request merged**